### PR TITLE
pdftk: reinit at 3.0.8

### PR DESCRIPTION
**Pull request by averelld**
###### Motivation for this change
I think this is basically a maintained rewrite of the previous package. See also https://gitlab.com/pdftk-java/pdftk#known-differences-with-pdftk

Independent testing would be good. I'm keeping maintainers as is, I hope that's ok. Also, supported platforms are a guess.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by 7c6f434c** [comment]
Diff looks fine. I wonder if it is a good idea to have both for some time (so that there is an easy way to the old one in case something rare is found).

The succesful build that prints a ton of Java compilation errors looks scary…

Basic functionality seems to work, I tried `burst`, `cat` with multiple files and with page selection and with rotation, `stamp`, `background`

**Comment by averelld** [comment]
> Diff looks fine. I wonder if it is a good idea to have both for some time (so that there is an easy way to the old one in case something rare is found).
>

I've added a copy as pdftk-legacy, maybe after the next release that can be removed. 

> The succesful build that prints a ton of Java compilation errors looks scary…

It worked because those wrongly encoded chars were only in comments, I've fixed the gradle call so it's not as scary now.

**Comment by 7c6f434c** [comment]
> I've added a copy as pdftk-legacy, maybe after the next release that can be removed.
Yes, and it could also be made `loPrio` (which is redundant but signals our low desire for its continued use).

> I've fixed the gradle call so it's not as scary now.

Have you updated the output hash, though?

(also, @grahamc is it even possible to start detecting this on ofBorg? first build left something with the old outputHash in the store, so the new build succeeded)

**Comment by averelld** [comment]
> > I've added a copy as pdftk-legacy, maybe after the next release that can be removed.
>  Yes, and it could also be made `loPrio` (which is redundant but signals our low desire for its continued use).
> 
I've added the lowPrio () wrapping
> > I've fixed the gradle call so it's not as scary now.
> 
> Have you updated the output hash, though?
That is not needed, because those comments have no effect on the output jar.
> 
> (also, @grahamc is it even possible to start detecting this on ofBorg? first build left something with the old outputHash in the store, so the new build succeeded)

Good point, also maybe now I'm breaking auto-updates via bot with this pattern?

**Comment by 7c6f434c** [comment]
> Good point, also maybe now I'm breaking auto-updates via bot with this pattern?

Hm, I think so. Maybe this fixed-output doesn't pay off in this case…

Also, maybe add youself as a maintainer?

**Comment by averelld** [comment]
As pointed out [here](https://github.com/NixOS/nixpkgs/pull/75546#discussion_r357792465), java-compiler upgrades would break this, so going back to the old perl-patchery stuff.

**Comment by Mic92** [comment]
I would just drop the old one. No need to have both available.

**Comment by Mic92** [comment]
Furthermore I would rename it to pdftk-java and add an alias from pdftk to pdftk-java in `pkgs/top-level/aliases.nix`

**Comment by 7c6f434c** [comment]
@Mic92 it is unmaintained but working and its domain area is not exactly changing match; the port's upstream says they are not yet fully confident in the port.

File: pkgs/tools/typesetting/pdftk/default.nix [hunk 1]
```diff
@@ -1,42 +1,95 @@
...
+        | sh
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "12b7lw1zpj69pv4bpbrm6pi0ip02ay3dfj3vcy2jyikfbwdb3qcz";
+  };
+
+  # Point to our local deps repo
+  gradleInit = writeText "init.gradle" ''
```
Comment (Mic92) [inline pkgs/tools/typesetting/pdftk/default.nix]: Maybe this gradle fetching could be factored out into a proper fetcher eventually. But this does not need to happen as part of this `PR`.

**Comment by Mic92** [comment]
@7c6f434c I would not call it `-legacy` in that case but keep the old name than.

**Comment by 7c6f434c** [comment]
Given this is a semiauto port, and `pdftk` requires `gcj` requiring old `gcc`, I prefer the middle road of moving the old version to legacy for some time.

The old `pdftk` definitely has not _accumulated_ any bugs in the last 6 years, there are of course a few cornercase bugs _discovered_ in the past 6 years, but the port hasn't yet fixed all of them, and has definitely added a few minor bugs of its own (some of those have been fixed already, but who knows how many remain).

**Pull request merged**